src/feature/hs/hs_config.c

```diff
@@ -471,25 +475,23 @@ config_generic_service(const config_line_t *line_,
   int have_dir_group_read = 0, have_max_streams = 0;
   int have_max_streams_close = 0;
 
-  tor_assert(line_);
+  tor_assert(line);
   tor_assert(options);
   tor_assert(service);
 
   /* Makes thing easier. */
   config = &service->config;
 
+  if (BUG(strcasecmp(line->key, "HiddenServiceDir")))
+    return -1;
+
   /* The first line starts with HiddenServiceDir so we consider what's next is
    * the configuration of the service. */
-  for (line = line_; line ; line = line->next) {
+  for ( ; line ; line = line->next) {
     int ok = 0;
 
     /* This indicate that we have a new service to configure. */
     if (!strcasecmp(line->key, "HiddenServiceDir")) {
-      /* This function only configures one service at a time so if we've
-       * already seen one, stop right now. */
-      if (dir_seen) {
-        break;
-      }
       /* Ok, we've seen one and we are about to configure it. */
       dir_seen = 1;
```

Comment:
Which also makes this `dir_seen` irrelevant.

Reply:
I don't see dir_seen any longer either.